core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -934,8 +934,11 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
       throw new ConfigException(s"Disabling the '${GroupType.CLASSIC}' protocol is not supported.")
     }
     if (protocols.contains(GroupType.CONSUMER)) {
```

Comment:
What if `group.coordinator.new.enable=true` to be used in ZK mode?

https://github.com/apache/kafka/blob/2d9994e0de915037525f041ff9a9b9325f838938/core/src/main/scala/kafka/server/KafkaConfig.scala#L944

Reply:
`group.coordinator.new.enable` is an internal config which is not supposed to be used, except in some tests. I can still strengthen the validation of it too if you want.

Reply:
> except in some tests. I can still strengthen the validation of it too if you want.

We don't need extra validation as it is used by testing only. BTW, will that internal config be removed in the future?

Reply:
Yes. We will make the new coordinator the default in 4.0 and remove the config. We may keep it until 4.1 in order to have a back door to disable it.